### PR TITLE
Add Nullable to Instrumenter API response args

### PR DESCRIPTION
**Pull request by trask**
@anuraaga @mateuszrzeszutek just confirming, did you want to call these methods with null response for symmetry, or alternatively can we skip calling them if there's no response (and then don't need to null check in the implementations)?

### CODE REVIEW

**Comment by trask** [comment]
I'm wondering if these particular methods should only be called with non-null response?

* statusCode()
* responseContentLength()
* responseContentLengthUncompressed()

File: instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/ArmeriaHttpAttributesExtractor.java [hunk 1]
```diff
@@ -53,28 +53,36 @@ protected String userAgent(RequestContext ctx) {
...
+    if (status.equals(HttpStatus.UNKNOWN)) {
+      return null;
     }
-    return null;
+    return (long) status.code();
   }
 
   @Override
-  protected String flavor(RequestContext ctx, RequestLog requestLog) {
+  protected String flavor(RequestContext ctx, @Nullable RequestLog requestLog) {
```
Comment (anuraaga) [inline instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/ArmeriaHttpAttributesExtractor.java:85]: Hmm - I'd like to not leak `@Nullable` in when it's not null. I guess what we want is to "only allow null if Void". But not sure we can model that, so maybe it's ok as it is, does it cause any nullability issues right now?

Comment (trask) [reply instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/ArmeriaHttpAttributesExtractor.java:85]: I think most instrumentations (other than armeria) will also pass `null` response to `end()` when exception occurs

Comment (anuraaga) [reply instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/ArmeriaHttpAttributesExtractor.java:85]: Ah thanks - in that case yeah skipping the invocation if null seems easier for instrumentation

Comment (trask) [reply instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/ArmeriaHttpAttributesExtractor.java:82]: added skipping of calls to

* statusCode()
* responseContentLength()
* responseContentLengthUncompressed()

when response is null

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpAttributesExtractor.java [hunk 2]
```diff
@@ -85,27 +90,81 @@ protected final void onEnd(AttributesBuilder attributes, REQUEST request, RESPON
...
-  protected abstract Long requestContentLength(REQUEST request, RESPONSE response);
-
+  protected abstract Long requestContentLength(REQUEST request, @Nullable RESPONSE response);
+
+  /**
+   * Extracts the {@code http.request_content_length_uncompressed} span attribute.
+   *
+   * <p>This is called from both {@link Instrumenter#start(Context, Object)} and {@link
+   * Instrumenter#end(Context, Object, Object, Throwable)}. When it is called from {@code start},
+   * {@code response} will be {@code null}.
```
Comment (mateuszrzeszutek) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpAttributesExtractor.java:108]: Those methods aren't currently called in `start()`

**Review by anuraaga — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**